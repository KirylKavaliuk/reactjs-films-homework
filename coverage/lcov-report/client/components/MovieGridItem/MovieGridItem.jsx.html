<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/MovieGridItem/MovieGridItem.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/MovieGridItem</a> MovieGridItem.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
&nbsp;
import { withDialogContext } from 'utils/dialog';
import withConditionalRendering from 'utils/rendering';
&nbsp;
import classNames from 'classnames';
&nbsp;
import Image from 'components/Image/Image';
import Icon from 'components/Icon/Icon';
import Button from 'components/Button/Button';
import Link from 'components/Link/Link';
&nbsp;
import styles from './MovieGridItem.scss';
&nbsp;
class MovieGridItem extends Component {
  state = {
    descriptionOpen: false,
  };
&nbsp;
  toggleDescriptionHandler = () =&gt; {
    const { descriptionOpen } = this.state;
    this.setState({ descriptionOpen: !descriptionOpen });
  }
&nbsp;
  shouldComponentUpdate = (nextProps, nextState) =&gt; (
    nextState.descriptionOpen !== this.state.descriptionOpen
  )
&nbsp;
  render() {
    const { descriptionOpen } = this.state;
    const { movie } = this.props;
&nbsp;
    return (
      &lt;div className={ styles.movieItem }&gt;
        &lt;Image db className={ styles.poster } width={ 1280 } src={ movie.poster_path }/&gt;
&nbsp;
        { !descriptionOpen &amp;&amp; &lt;div className={ styles.more }&gt;
          &lt;Link params={{ trailer: movie.id }}&gt;
            &lt;div className={ styles.playButton }&gt;
              &lt;Icon className={ styles.playButtonIcon } name='play'/&gt;
              &lt;h3 className={ styles.playButtonHeader }&gt;Play Now&lt;/h3&gt;
            &lt;/div&gt;
          &lt;/Link&gt;
&nbsp;
          &lt;Button transparent label='View Info' onClick={ this.toggleDescriptionHandler }/&gt;
        &lt;/div&gt; }
&nbsp;
        &lt;div className={ classNames(styles.info, { [styles.infoVisible]: descriptionOpen }) }&gt;
          &lt;div className={ classNames(
            styles.cancelIcon,
            { [styles.cancelIconVisible]: descriptionOpen },
          ) }
            onClick={ this.toggleDescriptionHandler }
          &gt;
            &lt;Icon name='cancel' /&gt;
          &lt;/div&gt;
&nbsp;
          &lt;header className={
            classNames(styles.header, { [styles.headerOffset]: descriptionOpen })
          }&gt;
            &lt;Link params={{ movie: movie.id }}&gt;
              &lt;h1 title={ movie.title } className={ styles.title }&gt;
                { movie.title }
              &lt;/h1&gt;
            &lt;/Link&gt;
            &lt;p className={ styles.genres }&gt;
              { movie.genres &amp;&amp; movie.genres.slice(0, 3).map((genre, index) =&gt; (
                &lt;Link
                  className={ styles.genre }
                  key={ index }
                  to={ `/genre/${genre.id}` }
                  clearParams={ ['query', 'movie'] }
                &gt;
                  { genre.name }
                &lt;/Link&gt;
              )) }
            &lt;/p&gt;
            &lt;div className={ styles.rating }&gt;
              { (movie.vote_average / 2).toPrecision(2) }
            &lt;/div&gt;
          &lt;/header&gt;
&nbsp;
          &lt;p className={
            classNames(
              styles.descriptionText,
              { [styles.descriptionTextVisible]: !descriptionOpen },
            )
          }&gt;
            { movie.overview }
          &lt;/p&gt;
&nbsp;
          &lt;Link params={{ trailer: movie.id }}&gt;
            &lt;Button label='Watch Now' className={ styles.watchButton }/&gt;
          &lt;/Link&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
MovieGridItem.defaultProps = {
  movie: null,
};
&nbsp;
MovieGridItem.propTypes = {
  movie: PropTypes.object,
};
&nbsp;
export default withDialogContext(withConditionalRendering(MovieGridItem, 'movie'));
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
