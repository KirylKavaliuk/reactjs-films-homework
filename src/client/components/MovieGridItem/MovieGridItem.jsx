import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withDialogContext } from 'utils/dialog';
import withConditionalRendering from 'utils/rendering';

import classNames from 'classnames';

import Image from 'components/Image/Image';
import Icon from 'components/Icon/Icon';
import Button from 'components/Button/Button';
import Link from 'components/Link/Link';

import styles from './MovieGridItem.scss';

class MovieGridItem extends Component {
  state = {
    descriptionOpen: false,
  };

  toggleDescriptionHandler = () => {
    const { descriptionOpen } = this.state;
    this.setState({ descriptionOpen: !descriptionOpen });
  }

  shouldComponentUpdate = (nextProps, nextState) => (
    nextState.descriptionOpen !== this.state.descriptionOpen
  )

  render() {
    const { descriptionOpen } = this.state;
    const { movie, genres } = this.props;
    // eslint-disable-next-line
    const { genre_ids } = movie;
    const infoClasses = classNames(
      styles.info,
      { [styles.infoVisible]: descriptionOpen },
    );

    const cancelClasses = classNames(
      styles.cancelIcon,
      { [styles.cancelIconVisible]: descriptionOpen },
    );

    const headerClasses = classNames(
      styles.header,
      { [styles.headerOffset]: descriptionOpen },
    );

    const descriptionClasses = classNames(
      styles.descriptionText,
      { [styles.descriptionTextVisible]: !descriptionOpen },
    );

    // eslint-disable-next-line
    const _genres = genre_ids
      .slice(0, 3)
      .map(genreId => genres.find(elem => elem.id === genreId));

    return (
      <div className={ styles.movieItem }>
        <Image
          db
          className={ styles.poster }
          width={ 300 }
          src={ movie.poster_path }
          alt={ movie.title }
        />

        { !descriptionOpen && <div className={ styles.more }>
          <Link params={{ trailer: movie.id }}>
            <div className={ styles.playButton }>
              <Icon className={ styles.playButtonIcon } name='play'/>
              <h3 className={ styles.playButtonHeader }>Play Now</h3>
            </div>
          </Link>

          <Button transparent label='View Info' onClick={ this.toggleDescriptionHandler }/>
        </div> }

        <div className={ infoClasses }>
          <div
            className={ cancelClasses }
            onClick={ this.toggleDescriptionHandler }
          >
            <Icon name='cancel' />
          </div>

          <header className={ headerClasses }>
            <Link params={{ movie: movie.id }}>
              <h1 title={ movie.title } className={ styles.title }>
                { movie.title }
              </h1>
            </Link>
            <p className={ styles.genres }>
              { _genres.map((genre, index) => (
                <Link
                  className={ styles.genre }
                  key={ index }
                  to={ `/genre/${genre.id}` }
                  clearParams={ ['query', 'movie'] }
                >
                  { genre.name }
                </Link>
              )) }
            </p>
            <div className={ styles.rating }>
              { (movie.vote_average / 2).toPrecision(2) }
            </div>
          </header>

          <p className={ descriptionClasses }>{ movie.overview }</p>

          <Link params={{ trailer: movie.id }}>
            <Button label='Watch Now' className={ styles.watchButton }/>
          </Link>
        </div>
      </div>
    );
  }
}

MovieGridItem.defaultProps = {
  movie: null,
};

MovieGridItem.propTypes = {
  movie: PropTypes.object,
};

export default withDialogContext(withConditionalRendering(MovieGridItem, 'movie'));
