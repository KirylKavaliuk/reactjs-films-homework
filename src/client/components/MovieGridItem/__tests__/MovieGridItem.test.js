import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import MovieGridItem from '../MovieGridItem';

it('<MovieGridItem/>', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <MovieGridItem
        movie={{
          id: '1',
          poster_path: '/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg',
          title: 'title',
          genres: [
            { id: 1, name: 'genre1' },
            { id: 2, name: 'genre2' },
            { id: 3, name: 'genre3' },
          ],
          vote_average: 3.5,
          overview: 'overview',
        }}
      />
    </MemoryRouter>,
  );

  const movieGridItem = tree.toJSON();
  expect(tree).not.toBeNull();
  expect(movieGridItem).toMatchSnapshot();

  const descButton = movieGridItem.children[1].children[1];

  descButton.props.onClick();
  expect(movieGridItem).toMatchSnapshot();
  expect(tree).toMatchSnapshot();
});

it('<MovieGridItem/>', () => {
  const tree = renderer.create(<MovieGridItem/>).toJSON();

  expect(tree).toBeNull();
  expect(tree).toMatchSnapshot();
});
