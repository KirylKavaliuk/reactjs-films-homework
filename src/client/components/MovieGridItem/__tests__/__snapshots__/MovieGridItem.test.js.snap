// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movie grid item correctly 1`] = `
<div
  className="MovieGridItem__movie-item___3Q67r"
>
  <img
    className="Image__image___2zFQr MovieGridItem__poster___wnVUx"
    src="https://image.tmdb.org/t/p/w1280//5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
  />
  <div
    className="MovieGridItem__more____ieLx"
  >
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div>
        <svg
          className="MovieGridItem__play-button-icon___qKloO"
          viewBox="0 0 58 58"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="29"
            cy="29"
            r="29"
          />
          <path
            d="M44 29L22 44V14z"
            fill="#fff"
          />
          <path
            d="M22 45a.999.999 0 0 1-1-1V14a.999.999 0 0 1 1.564-.826l22 15a1.001 1.001 0 0 1-.001 1.652l-22 15A1.002 1.002 0 0 1 22 45zm1-29.107v26.215L42.225 29 23 15.893z"
            fill="#fff"
          />
        </svg>
        <h3
          className="MovieGridItem__play-button-header___44D4Q"
        >
          Play Now
        </h3>
      </div>
    </a>
    <button
      className="Button__button___1nbn1 Button__transparent___3tNuU"
      onClick={[Function]}
    >
      View Info
    </button>
  </div>
  <div
    className="MovieGridItem__info___2nyBg"
  >
    <div
      className="MovieGridItem__cancel-icon___3NGEZ"
      onClick={[Function]}
    >
      <svg
        className=""
        viewBox="0 0 512.001 512.001"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M294.111 256.001L504.109 46.003c10.523-10.524 10.523-27.586 0-38.109-10.524-10.524-27.587-10.524-38.11 0L256 217.892 46.002 7.894C35.478-2.63 18.416-2.63 7.893 7.894s-10.524 27.586 0 38.109l209.998 209.998L7.893 465.999c-10.524 10.524-10.524 27.586 0 38.109 10.524 10.524 27.586 10.523 38.109 0L256 294.11l209.997 209.998c10.524 10.524 27.587 10.523 38.11 0 10.523-10.524 10.523-27.586 0-38.109L294.111 256.001z"
        />
      </svg>
    </div>
    <header
      className="MovieGridItem__header___Q5Ce4"
    >
      <a
        className="Link__link___-IU0S"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <h1
          className="MovieGridItem__title___8rf-J"
          title="title"
        >
          title
        </h1>
      </a>
      <p
        className="MovieGridItem__genres___12mKt"
      >
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre1
        </a>
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre2
        </a>
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre3
        </a>
      </p>
      <div
        className="MovieGridItem__rating___1fN4K"
      >
        1.8
      </div>
    </header>
    <p
      className="MovieGridItem__description-text___3Bcef undefined"
    >
      overview
    </p>
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="Button__button___1nbn1 MovieGridItem__watch-button___zHoyA"
        onClick={null}
      >
        Watch Now
      </button>
    </a>
  </div>
</div>
`;

exports[`renders movie grid item correctly 2`] = `
<div
  className="MovieGridItem__movie-item___3Q67r"
>
  <img
    className="Image__image___2zFQr MovieGridItem__poster___wnVUx"
    src="https://image.tmdb.org/t/p/w1280//5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
  />
  <div
    className="MovieGridItem__more____ieLx"
  >
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div>
        <svg
          className="MovieGridItem__play-button-icon___qKloO"
          viewBox="0 0 58 58"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="29"
            cy="29"
            r="29"
          />
          <path
            d="M44 29L22 44V14z"
            fill="#fff"
          />
          <path
            d="M22 45a.999.999 0 0 1-1-1V14a.999.999 0 0 1 1.564-.826l22 15a1.001 1.001 0 0 1-.001 1.652l-22 15A1.002 1.002 0 0 1 22 45zm1-29.107v26.215L42.225 29 23 15.893z"
            fill="#fff"
          />
        </svg>
        <h3
          className="MovieGridItem__play-button-header___44D4Q"
        >
          Play Now
        </h3>
      </div>
    </a>
    <button
      className="Button__button___1nbn1 Button__transparent___3tNuU"
      onClick={[Function]}
    >
      View Info
    </button>
  </div>
  <div
    className="MovieGridItem__info___2nyBg"
  >
    <div
      className="MovieGridItem__cancel-icon___3NGEZ"
      onClick={[Function]}
    >
      <svg
        className=""
        viewBox="0 0 512.001 512.001"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M294.111 256.001L504.109 46.003c10.523-10.524 10.523-27.586 0-38.109-10.524-10.524-27.587-10.524-38.11 0L256 217.892 46.002 7.894C35.478-2.63 18.416-2.63 7.893 7.894s-10.524 27.586 0 38.109l209.998 209.998L7.893 465.999c-10.524 10.524-10.524 27.586 0 38.109 10.524 10.524 27.586 10.523 38.109 0L256 294.11l209.997 209.998c10.524 10.524 27.587 10.523 38.11 0 10.523-10.524 10.523-27.586 0-38.109L294.111 256.001z"
        />
      </svg>
    </div>
    <header
      className="MovieGridItem__header___Q5Ce4"
    >
      <a
        className="Link__link___-IU0S"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <h1
          className="MovieGridItem__title___8rf-J"
          title="title"
        >
          title
        </h1>
      </a>
      <p
        className="MovieGridItem__genres___12mKt"
      >
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre1
        </a>
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre2
        </a>
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre3
        </a>
      </p>
      <div
        className="MovieGridItem__rating___1fN4K"
      >
        1.8
      </div>
    </header>
    <p
      className="MovieGridItem__description-text___3Bcef undefined"
    >
      overview
    </p>
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="Button__button___1nbn1 MovieGridItem__watch-button___zHoyA"
        onClick={null}
      >
        Watch Now
      </button>
    </a>
  </div>
</div>
`;

exports[`renders movie grid item correctly 3`] = `
<div
  className="MovieGridItem__movie-item___3Q67r"
>
  <img
    className="Image__image___2zFQr MovieGridItem__poster___wnVUx"
    src="https://image.tmdb.org/t/p/w1280//5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
  />
  <div
    className="MovieGridItem__info___2nyBg MovieGridItem__info-visible___NrFi3"
  >
    <div
      className="MovieGridItem__cancel-icon___3NGEZ MovieGridItem__cancel-icon-visible___2HExa"
      onClick={[Function]}
    >
      <svg
        className=""
        viewBox="0 0 512.001 512.001"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M294.111 256.001L504.109 46.003c10.523-10.524 10.523-27.586 0-38.109-10.524-10.524-27.587-10.524-38.11 0L256 217.892 46.002 7.894C35.478-2.63 18.416-2.63 7.893 7.894s-10.524 27.586 0 38.109l209.998 209.998L7.893 465.999c-10.524 10.524-10.524 27.586 0 38.109 10.524 10.524 27.586 10.523 38.109 0L256 294.11l209.997 209.998c10.524 10.524 27.587 10.523 38.11 0 10.523-10.524 10.523-27.586 0-38.109L294.111 256.001z"
        />
      </svg>
    </div>
    <header
      className="MovieGridItem__header___Q5Ce4 MovieGridItem__header-offset___1kN73"
    >
      <a
        className="Link__link___-IU0S"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <h1
          className="MovieGridItem__title___8rf-J"
          title="title"
        >
          title
        </h1>
      </a>
      <p
        className="MovieGridItem__genres___12mKt"
      >
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre1
        </a>
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre2
        </a>
        <a
          className="Link__link___-IU0S MovieGridItem__genre___EAwMy"
          href="/genre/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre3
        </a>
      </p>
      <div
        className="MovieGridItem__rating___1fN4K"
      >
        1.8
      </div>
    </header>
    <p
      className="MovieGridItem__description-text___3Bcef"
    >
      overview
    </p>
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="Button__button___1nbn1 MovieGridItem__watch-button___zHoyA"
        onClick={null}
      >
        Watch Now
      </button>
    </a>
  </div>
</div>
`;

exports[`renders movie grid item uncorrectly 1`] = `null`;
